code/__HELPERS/files.dm

```diff
@@ -95,3 +95,22 @@ GLOBAL_VAR_INIT(fileaccess_timer, 0)
 	fcopy(file, filename)
 	. = md5filepath(filename)
 	fdel(filename)
+
+/**
+ * Sanitizes the name of each node in the path.
+ *
+ * Im case you are wondering when to use this proc and when to use SANITIZE_FILENAME,
+ *
+ * You use SANITIZE_FILENAME to sanitize the name of a file [e.g. example.txt]
+ *
+ * You use sanitize_filepath sanitize the path of a file [e.g. root/node/example.txt]
+ *
+ * If you use SANITIZE_FILENAME to sanitize a file path things will break.
+ */
+/proc/sanitize_filepath(path, delimiter = "/")
+	. = ""
+	var/list/all_nodes = splittext(path, delimiter)
+	for(var/node in all_nodes)
+		if(.)
+			. += delimiter // Add the delimiter before each successive node.
+		. += SANITIZE_FILENAME(node)
```

Comment:
I really hate that this is needed in any way

Reply:
Yea, I have heard that comment already. Please tell me if you have a better idea.

Reply:
I don't. Just making a new sanitize seems overkill, and the whole situation is frustrating

Reply:
We had bitterer pills to swallow in the past. Plus we can always remove this at later time when a more acceptable solution is found.